### DISCUSSION TITLE
What does `beginTransaction` returning `null` mean?

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#7499] [General]
When I want to start a transaction, I expect `beginTransaction()` to start the transaction, or fail. But it is possible for its return value to be `null`. Subsequent usages of e.g. `rollbackTransaction()` does not take `null` as a value, and the example code uses `if transactionId` checks.

What does a `null` transaction ID mean? If I receive `null` for a transaction ID, should I early-abort?

**Comment by denolfe** [comment]
The base database adapter returns `null` for transactionIDs to indicate that transactions are not supported or not enabled.

Are you passing any additional options to your database adapter?

**Reply by tyteen4a03** [reply]
> The base database adapter returns `null` for transactionIDs to indicate that transactions are not supported or not enabled.

I think this should not be hidden from the developer and instead should throw an exception. From reading the types it's not clear why the transaction ID can be null.

> Are you passing any additional options to your database adapter?

I'm not.
